src/UserStories/UserStories.js

```diff
@@ -8,21 +8,48 @@ import {
   VolumeDown,
   VolumeOff,
 } from '@mui/icons-material';
-import { IconButton } from '@mui/material';
 import cx from 'classnames';
 import React, { useRef } from 'react';
 import { useHistory } from 'react-router';
 import { Link } from 'react-router-dom';
 import { Avatar } from '../Avatar/Avatar';
 import { StoryImage } from '../StoryImage/StoryImage';
-import { absoluteToRelativeDate } from '../utils';
+import { absoluteToRelativeDate, numberBetween } from '../utils';
 import CircularProgress from '@mui/material/CircularProgress';
 import './UserStories.css';
 import { StoryVideo } from '../StroyVideo/StoryVideo';
-import { elementWidth } from '../utils';
 
-function getX(el) {
-  return el?.offsetLeft;
+const STORY_ASPECT_RATIO = 0.562;
+const STORY_VERTICAL_MARGIN = 18;
+const TABLET_BREAKPOINT = 1540;
+const TABLET_STORY_RATIO = 0.59;
+/**
+ * borders are needed to activate next-prev nav buttons hover
+ */
+const STORY_BORDER_WIDTH = 46;
+
+function calculateStoryDimensions() {
+  const maxHeight = window.innerHeight - STORY_VERTICAL_MARGIN * 2;
+  let height, width;
+  if (window.innerWidth < window.innerHeight) {
+    // strangely, real IG connects the height of the story to the width of the screen
+    height = Math.min(window.innerWidth - STORY_VERTICAL_MARGIN * 2, maxHeight);
+    width = height * STORY_ASPECT_RATIO;
+  } else {
+    if (window.innerWidth < TABLET_BREAKPOINT) {
+      height = numberBetween(
+        TABLET_STORY_RATIO * window.innerWidth,
+        500,
+        maxHeight
+      );
+
+      width = height * STORY_ASPECT_RATIO;
+    } else {
+      height = maxHeight;
+      width = height * STORY_ASPECT_RATIO;
+    }
+  }
```

Comment:
Since the `width` is calculated the same way in every branch of these ifs and elses, I'd do the calculation in once place after the main if.